### DISCUSSION TITLE
Get the Prompt to display curr_branch with branch_color only if it's in a git repo

### DISCUSSION THREAD

**Discussion by WERTBON1789** [xonsh/xonsh#4873] [Q&A]
My current prompt looks like this: `'{BOLD_GREEN}[{user}@{hostname} {BOLD_BLUE}{cwd_base}{BOLD_GREEN}]{branch_color}{curr_branch: [{}]}{BOLD_GREEN}{prompt_end} {RESET}'`

So it only displays the the git branch if I'm in a git repo, if I'm in one it looks like this: `[user@host dir] [master]$`

and the color of the branch is right, but what I want it to look like is this: `[user@host dir:master]$`

So what I came up with is: `'{BOLD_GREEN}[{user}@{hostname} {BOLD_BLUE}{cwd_base}{curr_branch::{branch_color}{}}{BOLD_GREEN}]{prompt_end} {RESET}'`

but it looks like this now: `[user@host dir:{branch_color}master]$`

It seems like it can't really do anything with the branch_color there, but when I just put an color in the same place it works.
So how do I get the branch_color thing in that place to work and give me the right color

**Comment by gforsyth** [comment]
If I understand what you're looking for, I think you want this (with the `{branch_color}` field outside of the `{curr_branch}` field and with a `RESET` outside of it:

```
$PROMPT='{BOLD_GREEN}[{user}@{hostname} {BOLD_BLUE}{cwd_base}{branch_color}{curr_branch::{}}{BOLD_GREEN}]{prompt_end} {RESET}'
```
